Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -5,11 +5,24 @@
 #pragma once
 
 #include <string>
+#include <type_traits>
 
 #include "Common/Config/Enums.h"
 
 namespace Config
 {
+namespace detail
+{
+template <typename T, bool = std::is_enum<T>::value>
```

Comment:
There should probably be a comment explaining why this exists, as opposed to using std::conditional and/or std::underlying_type

Reply:
Done. I added a comment explaining why `detail::UnderlyingType` exists.

I also replaced this with:

```c++
using UnderlyingType = typename std::enable_if_t<std::is_enum<T>{}, std::underlying_type<T>>::type
```

as this seems to do the same thing but in a clearer and more concise way, I think.

Reply:
LGTM 👌, though I believe `typename` and `::type` at the end can be omitted, since the `_t` variant of enable_if is being used.

Reply:
`typename` and `::type` could be removed, but then later in the file I'd have to use `typename detail::UnderlyingType<Enum>::type` instead of just `detail::UnderlyingType<Enum>`.

I tried to omit `typename` and `::type` by using `std::underlying_type_t<T>` but sadly that doesn't seem to work.